### DISCUSSION TITLE
Ignore paths in functions

### DISCUSSION THREAD

**Discussion by jayhale** [nhost/nhost#1520] [Q&A] [answered]
Is there a way to ignore certain files or paths in the `/functions` directory?

E.g., I have a `codegen.ts` file that I don't want to be deployed as a function, but it always gets deployed.

**Comment by plmercereau** [comment, marked as answer]
You can prefix the file or directory name with an underscore

**Reply by jayhale** [reply]
@plmercereau thanks.  Relevant lines [here](https://github.com/nhost/functions/blob/db0797aaf0e0a309b2caba0917672e2fc0bb3372/server.ts#L39):

```typescript
const files = glob.sync('**/*.@(js|ts)', {
    cwd: functionsPath,
    ignore: [
      '**/node_modules/**', // ignore node_modules directories
      '**/_*/**', // ignore files inside directories that start with _
      '**/_*' // ignore files that start with _
    ]
  })
```

**Comment by elitan** [comment]
For reference, it's documented here: https://docs.nhost.io/serverless-functions#routing.

> You can prepend files and folders with an underscore (`_`) to prevent them from being treated as Serverless Functions and be turned into HTTP endpoints. This is useful if you have, for example, a utils file (`functions/_utils.js`) or a utils folder (`functions/_utils/<utils-files>.js`).
